blenderproc/python/utility/Utility.py

```diff
@@ -708,3 +710,45 @@ def default(self, o: Any):
             # Convert it to a list
             return o.tolist()
         return json.JSONEncoder.default(self, o)
+
+def fileno(file_or_fd: Union[int, IO]) -> int:
+    """ Returns the file descriptor of the given file. 
+
+    :param file_or_fd: Either a file or a file descriptor. If a file descriptor is given, it is returned directly.
+    :return: The file descriptor of the given file.
+    """
+    fd = getattr(file_or_fd, 'fileno', lambda: file_or_fd)()
+    if not isinstance(fd, int):
+        raise ValueError("Expected a file (`.fileno()`) or a file descriptor")
```

Comment:
Okay, I just spent 3 minutes understanding this one line. Can we change this to:
I think it is overly complicated to return a newly created lambda just to make it callable. 
```suggestion
    if hasattr(file_or_fd, 'fileno'):
        fd = file_or_fd.fileno()
    else:
        fd = file_or_fd
    if not isinstance(fd, int):
        raise ValueError("Expected a file (`.fileno()`) or a file descriptor")
```

Reply:
I agree, however isn't this the same logic as with dict.get()?
https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/consider-using-get.html

Reply:
It is not the default key that I object to. I object using a lambda to align it with a function call.

That lambda is just constructed to avoid writing more lines of code.

Reply:
Yes, I agree that is confusing